### PR TITLE
[Community] add paginated_fetch

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

**Comment by GuillaumeDSM** [comment]
up

File: octobot/community/supabase_backend/community_supabase_client.py [hunk 1]
```diff
@@ -493,14 +489,39 @@ async def _fetch_paginated_history(
...
-            fetched_elements = (await request.execute()).data
+            result = await request.execute()
+            fetched_elements = result.data
+            total_elements_count = total_elements_count or result.count   # don't change total count within iteration
             total_elements += fetched_elements
             if(
                 len(fetched_elements) == 0 or
                 len(fetched_elements) < max_size or
-                (max_size == 0 and len(fetched_elements) == total_elements_count)
+                (total_elements_count and max_size == 0 and len(fetched_elements) == total_elements_count)
```
Comment (Herklos) [inline octobot/community/supabase_backend/community_supabase_client.py:524]: :+1:

Comment (GuillaumeDSM) [reply octobot/community/supabase_backend/community_supabase_client.py]: I just updated this condition to make it easier to understand

Comment (Herklos) [reply octobot/community/supabase_backend/community_supabase_client.py]: :+1:

**Pull request merged**